homeassistant/components/zwave_js/discovery.py

```diff
@@ -1082,15 +1056,49 @@ class ZWaveDiscoverySchema:
         device_class_generic={"Thermostat"},
         primary_value=SWITCH_MULTILEVEL_CURRENT_VALUE_SCHEMA,
     ),
-    # lights
-    # primary value is the currentValue (brightness)
-    # catch any device with multilevel CC as light
-    # NOTE: keep this at the bottom of the discovery scheme,
-    # to handle all others that need the multilevel CC first
+    # Handle the different combinations of Binary Switch, Multilevel Switch and Color Switch
+    # to create switches and/or (colored) lights. The goal is to:
+    # - couple Color Switch CC with Multilevel Switch CC if possible
+    # - couple Color Switch CC with Binary Switch CC as the first fallback
+    # - use Color Switch CC standalone as the last fallback
+    #
+    # Multilevel Switch CC (+ Color Switch CC) -> Dimmable light with or without color support.
     ZWaveDiscoverySchema(
         platform=Platform.LIGHT,
         primary_value=SWITCH_MULTILEVEL_CURRENT_VALUE_SCHEMA,
     ),
+    # Binary Switch CC when Multilevel Switch and Color Switch CC exist -> On/Off switch, assign color to light entity instead
+    ZWaveDiscoverySchema(
+        platform=Platform.SWITCH,
```

Comment:
Please add a test for this case. Here's the diagnostics dump for Aeotec Smart Switch 7. Add the test under `test_discovery.py` and just test that the expected entity state is created (one switch and one light).

[aeotec_smart_switch_7.json](https://github.com/user-attachments/files/17309066/aeotec_smart_switch_7.json)